webapp/src/components/training/Trainer.vue

```diff
@@ -219,14 +271,24 @@ async function startTraining(distributed: boolean): Promise<void> {
     debug("while training: %o", e);
     return;
   } finally {
-    displayModelCaching.value = true // show model caching buttons again after training
-    trainingGenerator.value = undefined;
-    isTraining.value = false
+    cleanupTrainingSession()
   }
 
   toaster.success("Training successfully completed");
 }
 
+/**
+ * 
+ */
+async function cleanupTrainingSession() {
+  // check if the function has been initialized
+  if (cleanupTrainingSessionFn.value === undefined) return
+  // Calling the cleanup function returns a promise
+  // awaiting the promise notifies the network that we are disconnecting
+  await cleanupTrainingSessionFn.value()
```

Comment:
that might fire multiple times (TTCTTU). consider moving the cleanup function locally and setting the outside one to undef before the `await` point so that no other ~threads can pass the `undef` check

Reply:
Good catch!